charts/cortex/templates/store-gateway/store-gateway-statefulset.yaml

```diff
@@ -129,4 +130,152 @@ spec:
+  annotations:
+    {{- toYaml $.Values.store_gateway.annotations | nindent 4 }}
+spec:
+  replicas: {{ $.Values.store_gateway.replicas }}
+  selector:
+    matchLabels: {{ include "cortex-store-gateway.matchLabels" $ | nindent 6 }}
+{{- $.Values.topologyAware.zoneLabel | nindent 6 }}: {{ $availabilityCell }}
+  updateStrategy:
+    {{- toYaml $.Values.store_gateway.strategy | nindent 4 }}
+  serviceName: {{ template "cortex-store-gateway.headlessname-zone-aware" dict "Values" $.Values "Release" $.Release "Chart" $.Chart "AvailabilityCell" $availabilityCell }}
+  {{- if $.Values.store_gateway.persistentVolume.enabled }}
+  volumeClaimTemplates:
+    - metadata:
+        name: storage
+        {{- if $.Values.store_gateway.persistentVolume.annotations }}
+        annotations:
+          {{ toYaml $.Values.store_gateway.persistentVolume.annotations | nindent 10 }}
+        {{- end }}
+      spec:
+        {{- if $.Values.store_gateway.persistentVolume.storageClass }}
+        {{- if (eq "-" $.Values.store_gateway.persistentVolume.storageClass) }}
+        storageClassName: ""
+        {{- else }}
+        storageClassName: "{{ $.Values.store_gateway.persistentVolume.storageClass }}"
+        {{- end }}
+        {{- end }}
+        accessModes:
+          {{- toYaml $.Values.store_gateway.persistentVolume.accessModes | nindent 10 }}
+        resources:
+          requests:
+            storage: "{{ $.Values.store_gateway.persistentVolume.size }}"
+  {{- end }}
+  template:
+    metadata:
+      labels: {{ include "cortex-store-gateway.labels" $ | nindent 8 }}
+{{- $.Values.topologyAware.zoneLabel | nindent 8 }}: {{ $availabilityCell }}
+        {{- with $.Values.store_gateway.podLabels }}
+        {{- toYaml . | nindent 8 }}
+        {{- end }}
+      annotations:
+        checksum/config: {{ include "cortex.configChecksum" $ }}
+        {{- with $.Values.store_gateway.podAnnotations }}
+        {{- toYaml . | nindent 8 }}
+        {{- end }}
+    spec:
+      serviceAccountName: {{ $.Values.store_gateway.serviceAccount.name | default (include "cortex.serviceAccountName" $ ) }}
+      {{- if $.Values.store_gateway.priorityClassName }}
+      priorityClassName: {{ $.Values.store_gateway.priorityClassName }}
+      {{- end }}
+      securityContext: {{- toYaml $.Values.store_gateway.securityContext | nindent 8 }}
+      initContainers:
+        {{- toYaml $.Values.store_gateway.initContainers | nindent 8 }}
+      {{- if $.Values.image.pullSecrets }}
+      imagePullSecrets:
+      {{- range $.Values.image.pullSecrets }}
+        - name: {{ . }}
+      {{- end }}
+      {{- end }}
+      nodeSelector:
+        {{- toYaml $.Values.store_gateway.nodeSelector | nindent 8 }}
+      affinity:
+        nodeAffinity:
+          requiredDuringSchedulingIgnoredDuringExecution:
+            nodeSelectorTerms:
+            - matchExpressions:
+              - key: {{ $.Values.topologyAware.zoneLabel }}
```

Comment:
Again, we are using the same label name for both node selection and as part of the pod labels. Maybe we should not let the user chose the one on the pods and just let the user select node "zone" label to be used.